File: run_migration.py
import psycopg2

DB_HOST = "db.efhbyrhxtsadbqjsfogc.supabase.co"
DB_NAME = "postgres"
DB_USER = "postgres"
DB_PASSWORD = "+KjkpPVMr639E/n"
DB_PORT = "5432"

filename = 'supabase_migration.sql'

with open(filename, 'r', encoding='utf-8') as f:
    sql_content = f.read()

try:
    conn = psycopg2.connect(
        host=DB_HOST,
        database=DB_NAME,
        user=DB_USER,
        password=DB_PASSWORD,
        port=DB_PORT,
        sslmode='require'
    )
    conn.autocommit = True
    cursor = conn.cursor()
    cursor.execute(sql_content)
    print(f"[OK] {filename} executed successfully")
    cursor.close()
    conn.close()
except Exception as e:
    print(f"[FAIL] Error executing {filename}: {e}")

File: run_migration.js
const { Client } = require('pg');

const client = new Client({
  connectionString: 'postgresql://postgres:%2BKjkpPVMr639E%2Fn@db.efhbyrhxtsadbqjsfogc.supabase.co:5432/postgres',
  ssl: { rejectUnauthorized: false }
});

async function run() {
  try {
    await client.connect();
    
    const sql = `
      ALTER TABLE profiles
      ADD COLUMN IF NOT EXISTS role TEXT,
      ADD COLUMN IF NOT EXISTS bio TEXT,
      ADD COLUMN IF NOT EXISTS main_heroes TEXT[];
    `;
    
    await client.query(sql);
    console.log('[OK] Database updated successfully with profile columns');
  } catch (err) {
    console.error('[FAIL]', err);
  } finally {
    await client.end();
  }
}

run();
